### PR TITLE
last refactoring of terrain feature 

### PR DESCRIPTION
**Pull request by GGGuenni**
terrainFeature sadly can't be restricted any further. So the only thing I could do was making it private.

I added a small bug fix to this PR regarding aerialDistanceTo because I saw wrapped distance was not only be considered on wrapped worlds.

### CODE REVIEW

File: core/src/com/unciv/logic/map/MapUnit.kt [hunk 1]
```diff
@@ -392,19 +392,20 @@ class MapUnit {
...
                 }
                 if (tile.improvementInProgress == "Remove Road" || tile.improvementInProgress == "Remove Railroad")
                     tile.roadStatus = RoadStatus.None
                 else {
-                    // We put "tile.terrainFeature!=null" because of a strange edge case that SHOULD be solved from 3.11.11+, so we should remove it then and see
-                    if (tile.terrainFeature != null && tile.tileMap.gameInfo.ruleSet.terrains[tile.terrainFeature!!]!!.uniques
-                                    .contains("Provides a one-time Production bonus to the closest city when cut down"))
-                        tryProvideProductionToClosestCity()
-                    tile.terrainFeature = null
+                    val removedFeature = tile.terrainFeatures.find { tile.tileMap.gameInfo.ruleSet.terrains[it]!!.uniques
```
Comment (yairm210) [inline core/src/com/unciv/logic/map/MapUnit.kt:403]: I think there's a bit of a mix up
The removed feature should be the one specified in the name of the improvement, tile.improvementInProgress.replace("Remove ","") or something (removePrefix? I remember there's something like that but not sure in what language)
And then, we should check tryProvideProductionToClosestCity on that.

In the current code, if I have features A and B, and I have "Remove A" but B happens to have the unique, B will be removed. Also if I'm reading this correctly remove jungle marsh and fallout won't work?

File: core/src/com/unciv/logic/map/MapUnit.kt [hunk 2]
```diff
@@ -417,7 +418,7 @@ class MapUnit {
         tile.improvementInProgress = null
     }
 
-    private fun tryProvideProductionToClosestCity() {
+    private fun tryProvideProductionToClosestCity(cause: String) {
```
Comment (yairm210) [inline core/src/com/unciv/logic/map/MapUnit.kt:421]: For better readability call this removedTerrainFeature or something, "cause" is very vague

File: core/src/com/unciv/logic/map/TileInfo.kt [hunk 3]
```diff
@@ -436,10 +417,13 @@ open class TileInfo {
...
-        val otherTileUnwrappedPos = tileMap.getUnWrappedPosition(otherTile.position)
-        val xDeltaWrapped = position.x - otherTileUnwrappedPos.x
-        val yDeltaWrapped = position.y - otherTileUnwrappedPos.y
-        val wrappedDistance = listOf(abs(xDeltaWrapped), abs(yDeltaWrapped), abs(xDeltaWrapped - yDeltaWrapped)).maxOrNull()!!
+        var wrappedDistance = Float.MAX_VALUE
+        if (tileMap.mapParameters.worldWrap) {
+            val otherTileUnwrappedPos = tileMap.getUnWrappedPosition(otherTile.position)
+            val xDeltaWrapped = position.x - otherTileUnwrappedPos.x
+            val yDeltaWrapped = position.y - otherTileUnwrappedPos.y
+            wrappedDistance = listOf(abs(xDeltaWrapped), abs(yDeltaWrapped), abs(xDeltaWrapped - yDeltaWrapped)).maxOrNull()!!
```
Comment (yairm210) [inline core/src/com/unciv/logic/map/TileInfo.kt:425]: max(a,b,c) should work, no? It's more directly readable IMHO and that way you don't need to create an extra list

File: core/src/com/unciv/logic/map/MapUnit.kt (hunk 1 above)
Comment (GGGuenni) [reply core/src/com/unciv/logic/map/MapUnit.kt]: Yea you are right
done

**Pull request merged**